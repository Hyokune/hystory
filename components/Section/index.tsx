import React from 'react'
import cx from 'classnames'

interface ISection {
  children: JSX.Element[] | JSX.Element,
  fullHeight?: boolean
}

class Section extends React.Component<ISection> {
  render() {
    const { fullHeight } = this.props

    return (
      <section className={cx({
        "section": true,
        "full-height": fullHeight
      })}>
        {this.props.children}
      </section>
    )
  }
}

export default Section
