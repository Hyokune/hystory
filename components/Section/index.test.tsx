import { shallow } from 'enzyme'
import React from 'react'
import Section from './index'

const childElement = <p>Child element</p>

describe('Section component', () => {
  test('render component without error', () => {
    const wrapper = shallow(<Section>{childElement}</Section>)
    expect(wrapper).toMatchSnapshot()
  })
})
