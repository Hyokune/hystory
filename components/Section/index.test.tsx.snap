// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component render component without error 1`] = `
<section
  className=""
>
  <p>
    Child element
  </p>
</section>
`;
